metadata-integration/java/datahub-client/src/main/java/datahub/client/patch/datajob/DataJobInputOutputPatchBuilder.java

```diff
@@ -0,0 +1,115 @@
+import com.linkedin.common.urn.DatasetUrn;
+import com.linkedin.common.urn.Urn;
+import datahub.client.patch.AbstractMultiFieldPatchBuilder;
+import datahub.client.patch.PatchOperationType;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeFactory.*;
+import static com.linkedin.metadata.Constants.*;
+
+
+public class DataJobInputOutputPatchBuilder extends AbstractMultiFieldPatchBuilder<DataJobInputOutputPatchBuilder> {
+  private static final String INPUT_DATA_JOB_EDGES_PATH_START = "/inputDatajobEdges/";
+  private static final String INPUT_DATASET_EDGES_PATH_START = "/inputDatasetEdges/";
+  private static final String OUTPUT_DATASET_EDGES_PATH_START = "/outputDatasetEdges/";
+
+  private static final String DESTINATION_URN_KEY = "destinationUrn";
+  private static final String LAST_MODIFIED_KEY = "lastModified";
+  private static final String CREATED_KEY = "created";
+
+  private static final String INPUT_DATASET_FIELDS_PATH_START = "/inputDatasetFields/";
+  private static final String OUTPUT_DATASET_FIELDS_PATH_START = "/outputDatasetFields/";
+
+  private static final String TIME_KEY = "time";
+  private static final String ACTOR_KEY = "actor";
+
+  public DataJobInputOutputPatchBuilder addInputDatajobEdge(@Nonnull DataJobUrn dataJobUrn) {
+    ObjectNode value = createEdgeValue(dataJobUrn);
+
+    pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), INPUT_DATA_JOB_EDGES_PATH_START + dataJobUrn, value));
+    return this;
+  }
+
+  public DataJobInputOutputPatchBuilder removeInputDatajobEdge(@Nonnull DataJobUrn dataJobUrn) {
+    pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(), INPUT_DATA_JOB_EDGES_PATH_START + dataJobUrn, null));
+    return this;
+  }
+
+  public DataJobInputOutputPatchBuilder addInputDatasetEdge(@Nonnull DatasetUrn datasetUrn) {
+    ObjectNode value = createEdgeValue(datasetUrn);
+
+    pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), INPUT_DATASET_EDGES_PATH_START + datasetUrn, value));
+    return this;
+  }
+
+  public DataJobInputOutputPatchBuilder removeInputDatasetEdge(@Nonnull DatasetUrn datasetUrn) {
+    pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(), INPUT_DATASET_EDGES_PATH_START + datasetUrn, null));
+    return this;
+  }
+
+  public DataJobInputOutputPatchBuilder addOutputDatasetEdge(@Nonnull DatasetUrn datasetUrn) {
+    ObjectNode value = createEdgeValue(datasetUrn);
+
+    pathValues.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), OUTPUT_DATASET_EDGES_PATH_START + datasetUrn, value));
+    return this;
+  }
+
+  public DataJobInputOutputPatchBuilder removeOutputDatasetEdge(@Nonnull DatasetUrn datasetUrn) {
+    pathValues.add(ImmutableTriple.of(PatchOperationType.REMOVE.getValue(), OUTPUT_DATASET_EDGES_PATH_START + datasetUrn, null));
+    return this;
+  }
+
+  private ObjectNode createEdgeValue(Urn urn) {
```

Comment:
This forces the timestamp for the added edge to be current time, which doesn't work if the producer wants to set a different time (e.g. processing an audit log and finding a job that was run a day ago). We need to expand the API to support passing in the full edge as well. Perhaps there can be a default implementation like this as a shortcut for people who don't care.

Reply:
Sure, will add it